### PR TITLE
OH2-252 | Fix audit logging from API

### PR DESCRIPTION
**Pull request by ArnaudFonzam**
See OH2-252.

- [x] add UserSession class
- [x] add also all the class that are use in UserSession class like : DelayTimer
- [x] update getCurrentUser() method
- [x] add property updatable = false in all the model class to avoid updating createBy and createDate column

### CODE REVIEW

File: src/main/java/org/isf/sessionaudit/model/UserSession.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package org.isf.sessionaudit.model;
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/sessionaudit/model/UserSession.java:1]: Missing license header.

Comment (mwithi) [reply src/main/java/org/isf/sessionaudit/model/UserSession.java:1]: True, in GUI it was there

File: src/main/java/org/isf/sessionaudit/model/UserSession.java
Comment (mwithi) [inline src/main/java/org/isf/sessionaudit/model/UserSession.java:1]: We need a PR on GUI where we remove this class from there

Comment (mwithi) [reply src/main/java/org/isf/sessionaudit/model/UserSession.java:1]: or we extend it adding the `restartSession()` and override `setTimer()` with `org.isf.utils.jobjects.DelayTimer` as parameter

File: src/main/java/org/isf/sessionaudit/model/UserSession.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import java.util.Map;
+
+import org.isf.menu.model.User;
+import org.isf.utils.time.DelayTimer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class UserSession {
+
+	private static final int LOGIN_FAILED = 2;
```
Comment (mwithi) [inline src/main/java/org/isf/sessionaudit/model/UserSession.java:13]: Not used

**Review by mwithi — DISMISSED** [review]
LGTM. Thanks a lot!

File: src/main/java/org/isf/menu/manager/UserBrowsingManager.java [hunk 3]
```diff
@@ -48,7 +48,9 @@ public class UserBrowsingManager {
 	private MenuIoOperations ioOperations;
 
 	public static String getCurrentUser() {
-		return MDC.get("OHUser");
+		if (UserSession.getUser() != null)
+			return UserSession.getUser().getUserName();
```
Comment (mwithi) [inline src/main/java/org/isf/menu/manager/UserBrowsingManager.java:52]:
```suggestion
		if (UserSession.getUser() != null) {
			return UserSession.getUser().getUserName();
		}
```

File: src/main/java/org/isf/utils/db/DefaultAuditorAwareImpl.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package org.isf.utils.db;
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/utils/db/DefaultAuditorAwareImpl.java:1]: Missing the license header.

**Review by dbmalkovsky — DISMISSED** [review]
LGTW, after adding the license header

File: src/main/java/org/isf/utils/db/DefaultAuditorAwareImpl.java (hunk 4 above)
Comment (mwithi) [reply src/main/java/org/isf/utils/db/DefaultAuditorAwareImpl.java]: Done, thanks!

**Pull request merged**